Gem/Connections/Services/WalletConnectorInteractor.swift

```diff
@@ -1,31 +1,95 @@
 // Copyright (c). Gem Wallet. All rights reserved.
 
-import Foundation
-import WalletConnector
+import SwiftUI
 import Primitives
+import WalletConnector
+
+@Observable
+final class WalletConnectorInteractor {
+    var isPresentingError: String? = nil
+    var isPresentingConnectionBar: Bool = false
+    var isPresentingSheeet: WalletConnectorSheetType? = nil
+
+    init() {}
+
+    func cancel(type: WalletConnectorSheetType) {
+        let error = ConnectionsError.userCancelled
+        switch type {
+        case .transferData(let transferDataCallback):
```

Comment:
these 3 case let looks like can be merged?

Reply:
unfortunately, we can't use Pattern variable bound right here like this:
```
switch type {
        case .transferData(let callback),
                .signMessage(let callback),
                .connectionProposal(let callback):
                callback.delegate(.failure(error))
        }
```        
case expected to be 1 type, but we using generics inside